src/lib/features/project/project-owners-read-model.ts

```diff
@@ -0,0 +1,151 @@
+        roleId: number,
+    ): Promise<Record<string, UserProjectOwner[]>> {
+        const usersResult = await this.db
+            .select(
+                'user.username',
+                'user.name',
+                'user.email',
+                'user.image_url',
+                'ru.created_at',
+                'ru.project',
+            )
+            .from(`${T.ROLE_USER} as ru`)
+            .join(`${T.ROLES} as r`, 'ru.role_id', 'r.id')
+            .where('r.id', roleId)
+            .join(`${T.USERS} as user`, 'ru.user_id', 'user.id');
+        const usersDict: Record<string, UserProjectOwner[]> = {};
+
+        usersResult.forEach((user) => {
+            const project = user.project as string;
+
+            const data: UserProjectOwner = {
+                ownerType: 'user',
+                name: user?.name || user?.username,
+                email: user?.email,
+                imageUrl: user?.image_url,
+            };
+
+            if (project in usersDict) {
+                usersDict[project] = [...usersDict[project], data];
+            } else {
+                usersDict[project] = [data];
+            }
+        });
+
+        return usersDict;
+    }
+
+    private async getAllProjectGroupsByRole(
+        roleId: number,
+    ): Promise<Record<string, GroupProjectOwner[]>> {
+        const groupsResult = await this.db
+            .select('groups.name', 'gr.created_at', 'gr.project')
+            .from(`${T.GROUP_ROLE} as gr`)
+            .join(`${T.ROLES} as r`, 'gr.role_id', 'r.id')
+            .where('r.id', roleId)
+            .join('groups', 'gr.group_id', 'groups.id');
+
+        const groupsDict: Record<string, GroupProjectOwner[]> = {};
+
+        groupsResult.forEach((group) => {
+            const project = group.project as string;
+
+            const data: GroupProjectOwner = {
+                ownerType: 'group',
+                name: group?.name,
+            };
+
+            if (project in groupsDict) {
+                groupsDict[project] = [...groupsDict[project], data];
+            } else {
+                groupsDict[project] = [data];
+            }
+        });
+
+        return groupsDict;
+    }
+
+    async getAllProjectOwners(): Promise<ProjectOwnersDictionary> {
+        const ownerRole = await this.roleStore.getRoleByName(RoleName.OWNER);
+        const usersDict = await this.getAllProjectUsersByRole(ownerRole.id);
+        const groupsDict = await this.getAllProjectGroupsByRole(ownerRole.id);
+
+        const projects = [
+            ...new Set([...Object.keys(usersDict), ...Object.keys(groupsDict)]),
+        ];
+
+        const dict = Object.fromEntries(
+            projects.map((project) => {
+                return [
+                    project,
+                    [
+                        ...(usersDict[project] || []),
+                        ...(groupsDict[project] || []),
+                    ],
+                ];
+            }),
+        );
```

Comment:
This is optional, but there's a lot of spreading here. We're not necessarily doing it in a loop, but it's a lot more iterations than we need. How about doing something like this instead (coded in github, so might not work directly 😅 ):

```suggestion
        for (const [project, groups] of Object.entries(groupsDict) {
          if (project in usersDict) {
            usersDict[project] = usersDict[project].concat(groups)
          } else {
            usersDict[project] = groups
          }
        }
```

and then return the `usersDict` instead? That'll save us a lot of iteration and feels more readable to me 🤷🏼

Reply:
done in next PR